### PR TITLE
Provide Compatibliy for default Softmax axis

### PR DESCRIPTION
**Pull request by ghostplant**
This fix is to provide compatibility for original Keras backend interface.

NOTE: TF1.5 is using CUDA9 and CUDNN7 by default while most machines
keep CUDA8 and CUDNN6.

Keras cannot launch TF<=1.4 correct that makes all models fail, e.g: 
https://github.com/keras-team/keras/issues/9621

Signed-off-by: CUI Wei <ghostplant@qq.com>

### CODE REVIEW

**Comment by fchollet** [comment]
The latest version of TF is 1.7. It is thus reasonable to only support 1.5 and higher for new features.

We are not breaking any code since `axis` in softmax is a new feature. It is important that simply upgrading Keras (without upgrading TF) should not break your Keras code, at least until you are ~3-4 versions late. This is the case here.
